lib/rubygems/config_file.rb

```diff
@@ -336,6 +336,10 @@ def set_api_key host, api_key
     load_api_keys # reload
   end
 
+  def unset_api_key!
+    File.exist?(credentials_path) ? File.delete(credentials_path) : false
```

Comment:
Agreed, but using `unless` might create confusion and hamper readability in my opinion? How about writing it in formal `if` - `else` format?